warehouse/packaging/models.py

```diff
@@ -690,12 +690,31 @@ def urls(self):
         return _urls
 
     def urls_by_verify_status(self, verified: bool):
-        matching_urls = {
+        verified_urls = {
             release_url.url
             for release_url in self._project_urls.values()  # type: ignore[attr-defined]
-            if release_url.verified == verified
+            if release_url.verified
         }
 
+        if verified:
+            matching_urls = verified_urls
+        else:
+            matching_urls = {
+                release_url.url
+                for release_url in self._project_urls.values()  # type: ignore[attr-defined] # noqa: E501
+                if not release_url.verified
+            }
+            # The Homepage and Download URLs in the release metadata are currently not
+            # verified, so we return them if the user requests non-verified URLs *and*
+            # they are not verified in `project_urls`.
+            matching_urls.update(
+                {
+                    url
+                    for url in (self.home_page, self.download_url)
+                    if url is not None and url not in verified_urls
+                }
+            )
```

Comment:
Just summarizing to make sure I follow:

* If the context requests verified URLs `urls_by_verify_status(True)`, we return only URLs from `_project_urls` that are marked as verified.
* If the context requests unverified URLs `urls_by_verify_status(False)`, we return unverified URLs from `_project_urls`, *merged with* the free-standing `Release` URLs (`home_page` and `download_url`) *if* those URLs aren't in the verified set.

Is that correct? If so, this LGTM!

Reply:
Yes, that is correct!